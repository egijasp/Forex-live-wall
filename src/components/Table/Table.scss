.container {
  display: grid;
  justify-content: center;
  grid-template-columns: minmax(300px, 900px);
}

.table {
  border-collapse: collapse;
  width: 100%;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
  font-size: 12px;

  &--heading {
    text-align: center;
    background-color: #800080;
    font-weight: bold;
    letter-spacing: 2px;
    text-transform: uppercase;
    font-size: 13px;
  }
}

.table__cell {
  padding: 10px;
}

@media only screen and ( min-width: 600px ) {
  .table {
    font-size: 14px;

    &--heading {
      font-size: 15px;
    }
  }
}

@media (min-width: 768px) {
  .table {
    font-size: 15px;

    &--heading {
      font-size: 16px;
    }
  }
}
