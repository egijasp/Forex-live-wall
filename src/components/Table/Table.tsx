import { FC } from 'react';
import './Table.scss';
import { CurrencyData } from '../../helpers/types';

type TableProps = {
  data: CurrencyData[]
}

const Table: FC<TableProps> = ({ data }) => (
  <div className="container">
    <table className="table">
      <thead>
        <tr className="table--heading">
          <th className="table__cell">Pair</th>
          <th className="table__cell">BID</th>
          <th className="table__cell">ASK</th>
          <th className="table__cell">Changes %</th>
        </tr>
      </thead>
      <tbody>
        {data && data.map(({
          ticker, bid, ask, changes,
        }) => (
          <tr
            key={bid}
          >
            <td
              className="table__cell"
            >
              {ticker}
            </td>
            <td className="table__cell">{bid}</td>
            <td className="table__cell">{ask}</td>
            <td
              className="table__cell"
              style={{ color: changes > 0 ? 'darkgreen' : 'darkred' }}
            >
              {changes && changes.toFixed(3)}
              %
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

export default Table;
