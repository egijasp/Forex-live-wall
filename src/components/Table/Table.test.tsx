import renderer from 'react-test-renderer';
import Table from './Table';

const testData = [{
  ticker: 'EUR/USD',
  bid: '1.14154',
  ask: '1.14154',
  changes: -0.0035526924520561257,
}, {
  ticker: 'USD/JPY',
  bid: '114.185',
  ask: '114.185',
  changes: -8.757641041850795E-6,
},
];

it('renders correctly', () => {
  const component = renderer
    .create(<Table data={testData} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
