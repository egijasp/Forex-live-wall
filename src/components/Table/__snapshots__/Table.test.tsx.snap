// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <table
    className="table"
  >
    <thead>
      <tr
        className="table--heading"
      >
        <th
          className="table__cell"
        >
          Pair
        </th>
        <th
          className="table__cell"
        >
          BID
        </th>
        <th
          className="table__cell"
        >
          ASK
        </th>
        <th
          className="table__cell"
        >
          Changes %
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="table__cell"
        >
          EUR/USD
        </td>
        <td
          className="table__cell"
        >
          1.14154
        </td>
        <td
          className="table__cell"
        >
          1.14154
        </td>
        <td
          className="table__cell"
          style={
            Object {
              "color": "darkred",
            }
          }
        >
          -0.004
          %
        </td>
      </tr>
      <tr>
        <td
          className="table__cell"
        >
          USD/JPY
        </td>
        <td
          className="table__cell"
        >
          114.185
        </td>
        <td
          className="table__cell"
        >
          114.185
        </td>
        <td
          className="table__cell"
          style={
            Object {
              "color": "darkred",
            }
          }
        >
          -0.000
          %
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
